services/users/src/routes/team.ts

```diff
@@ -0,0 +1,102 @@
+import express from "express";
+import { asyncHandler } from "@api/common/src/middleware";
+import { BadRequestError } from "@api/common/src/errors";
+
+import { TeamModel, Team } from "../models/team";
+
+export const teamRoutes = express.Router();
+
+teamRoutes.route("/").post(
+  asyncHandler(async (req, res) => {
+    const { name, event, members, description, publicTeam } = req.body;
+
+    const team = await TeamModel.findOne({ name, event });
+
+    if (team) {
+      throw new BadRequestError("Team with this name already exists!");
+    }
+
+    const teams = await TeamModel.find({ event, members: req.user?.uid });
+
+    if (teams.length !== 0) {
+      throw new BadRequestError("User is already in a team for this event!");
+    }
+
+    await TeamModel.create({
+      name,
+      event,
+      members: [req.user?.uid, ...(members || [])],
+      description,
+      public: publicTeam,
+    });
+
+    res.status(200).send("Team created!");
+  })
+);
+
+teamRoutes.route("/").get(
+  asyncHandler(async (req, res) => {
+    const { event } = req.body;
```

Comment:
In a GET request, you can't use the req.body. As such, you 'll need to make this a query param.